get-started/nodejs/build-images.md

```diff
@@ -0,0 +1,244 @@
+
+```shell
+$ node server.js
+```
+
+To test that the application is working properly, we’ll first POST some JSON to the API and then make a GET request to see that the data has been saved. Open a new terminal and run the following curl commands:
+
+```shell
+$ curl --request POST \
+  --url http://localhost:8000/test \
+  --header 'content-type: application/json' \
+  --data '{
+	"msg": "testing"
+}'
+{"code":"success","payload":[{"msg":"testing","id":"31f23305-f5d0-4b4f-a16f-6f4c8ec93cf1","createDate":"2020-08-28T21:53:07.157Z"}]}
+
+$ curl http://localhost:8000/test
+{"code":"success","meta":{"total":1,"count":1},"payload":[{"msg":"testing","id":"31f23305-f5d0-4b4f-a16f-6f4c8ec93cf1","createDate":"2020-08-28T21:53:07.157Z"}]}
+```
+
+Switch back to the terminal where our server is running. You should now see the following requests in the server logs.
+
+```node
+2020-XX-31T16:35:08:4260  INFO: POST /test
+2020-XX-31T16:35:21:3560  INFO: GET /test
+```
+
+## Create a Dockerfile for Node.js
+
+A Dockerfile is a text document that contains all the commands a user could call on the command line to assemble an image. When we tell Docker to build our image by executing the `docker run` command, Docker reads these instructions and executes them one by one and creates a Docker image as a result.
+
+Let’s walk through the process of creating a Dockerfile for our application. In the root of your working directory, create a file named `Dockerfile` and open this file in your text editor.
+
+> **Note**
+>
+> The name of the Dockerfile is not important but the default filename for many commands is simply `Dockerfile`. So, we’ll use that as our filename throughout this series.
+
+The first thing we need to do is to add a line in our Dockerfile that tells Docker what base image we would like to use for our application.
+
+```dockerfile
+FROM node:12.18.1
+```
+
+Docker images can be inherited from other images. Therefore, instead of creating our own base image, we’ll use the official Node.js image that already has all the tools and packages that we need to run a Node.js application. You can think of this in the same way you would think about class inheritance in object oriented programming. For example, if we were able to create Docker images in JavaScript, we might write something like the following.
+
+`class MyImage extends NodeBaseImage {}`
+
+This would create a class called `MyImage` that inherited functionality from the base class `NodeBaseImage`.
+
+In the same way, when we use the `FROM` command, we tell Docker to include in our image all the functionality from the `node:12.18.1` image.
+
+The `NODE_ENV` environment variable specifies the environment in which an application is running (usually, development or production). One of the simplest things you can do to improve performance is to set `NODE_ENV` to `production`.
+
+```dockerfile
+ENV NODE_ENV=production
+```
+
+> **Note**
```

Comment:
The Note is part of the FROM section. Sorry I probably selected the wrong line for my first suggestion about adding the ENV

![nodejs](https://user-images.githubusercontent.com/207759/94134667-99466900-fe62-11ea-9ae2-be025ef6c861.jpg)